api.yml

```diff
@@ -19,9 +19,97 @@ paths:
     post:
       tags:
         - internal
-      summary: Verifies a credential and returns its verification status.
+      summary: Verify a given credential.
       operationId: verifyCredential
-      description: Verifies a credential and returns its verification status. Support of this part of the API is REQUIRED for implementations.
+      description: Performs a series of verification checks on a provided Verifiable Credential and returns the overall status of the checks as well as the status of each individual check. Support of this part of the API is REQUIRED for implementations.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              description: The credential to be verified and a set of options to use when verifying the credential.
```

Comment:
Certain checks can only be performed if we add an additional parameter, which is the proof request against which verification is being performed. Simple and important example: we ask Alice to show us her commercial driver's license. She shows us a non-commercial driver's license instead. The non-commercial driver's license will likely verify as far as its internal `proof` is concerned, but it doesn't match the criteria. Hackers could drive a truck through that gap.

So, do we want to allow an extra, optional parameter which is the criteria we're supposed to be satisfying? I know that raises all kinds of problems (especially, that we don't have a standard way to express criteria). But this feels like a fairly important omission.

Reply:
> So, do we want to allow an extra, optional parameter which is the criteria we're supposed to be satisfying? 

Yes, we want to do that... and I agree with you that we don't have a standard way to express that criteria yet. I've raised issue #8 to track this and ensure we address it in a later PR.